x/sync/manager.go

```diff
@@ -148,14 +150,19 @@ type ManagerConfig struct {
 	StateSyncNodes        []ids.NodeID
 }
 
-func NewManager(config ManagerConfig, registerer prometheus.Registerer) (*Manager, error) {
+func NewManager[R, C Proof](
```

Comment:
nit: Can we be consistent w/ the naming for generic types? We're using `TRange` on the struct definition but `R` on this constructor. I don't have a preference for us using `TRange` or `R` (I'd probably use `T` and `U`, or `R` and `C`, but all are fine to use here).